src/main/java/com/blackducksoftware/integration/hub/alert/channel/DistributionChannel.java

```diff
@@ -52,17 +53,20 @@ public CommonDistributionRepository getCommonDistributionRepository() {
     }
 
     public G getGlobalConfigEntity() {
-        final List<G> globalConfigs = globalRepository.findAll();
-        if (globalConfigs.size() == 1) {
-            return globalConfigs.get(0);
+        if (globalConfigEntity == null) {
+            final List<G> globalConfigs = globalRepository.findAll();
+            if (globalConfigs.size() == 1) {
+                globalConfigEntity = globalConfigs.get(0);
+            }
+            logger.error("Global Config did not have the expected number of rows: Expected 1, but found {}.", globalConfigs.size());
         }
-        logger.error("Global Config did not have the expected number of rows: expected 1, but found {}.", globalConfigs.size());
-        return null;
+
+        return globalConfigEntity;
     }
 
     public abstract void sendMessage(final E event, final C config);
 
-    public abstract String testMessage(final G globalConfig) throws IntegrationException;
```

Comment:
The reason we want to pass an entity here is because a test happens before anything gets saved. I'm thinking we should revisit this.

Reply:
Maybe this should take the distribution config. The test concept changes if we have a global config and a distribution config.